src/inference/src/os/lin/lin_system_conf.cpp

```diff
@@ -280,4 +280,84 @@ int getNumberOfCPUCores(bool bigCoresOnly) {
     return phys_cores;
 }
 
+std::vector<std::vector<int>> getNumOfAvailableCPUCores() {
+    std::vector<std::vector<int>> proc_type_table;
+    std::vector<int> all_table;
+    if (cpu._sockets == 1) {
+        proc_type_table.resize(1, std::vector<int>(PROC_TYPE_TABLE_SIZE, 0));
+    } else {
+        proc_type_table.resize(2, std::vector<int>(PROC_TYPE_TABLE_SIZE, 0));
+    }
+    all_table.resize(PROC_TYPE_TABLE_SIZE, 0);
+    for (int i = 0; i < cpu._processors; i++) {
```

Comment:
Could we just check `cpu._cpu_mapping_table[i][CPU_MAP_USED_FLAG] == -1` and count number of each type. Then reduce the number of each type from `cpu._proc_type_table`?

Reply:
Yes, the logic could be simplified.